pkg/ottl/ottlfuncs/func_is_match.go

```diff
@@ -32,11 +33,18 @@ func IsMatch[K any](target ottl.Getter[K], pattern string) (ottl.ExprFunc[K], er
 		if err != nil {
 			return nil, err
 		}
-		if val != nil {
-			if valStr, ok := val.(string); ok {
-				return compiledPattern.MatchString(valStr), nil
-			}
+
+		switch v := val.(type) {
```

Comment:
Should we actually add a "String" func instead, see ["Int"](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/ottlfuncs/func_int.go#L24)?

Reply:
@bogdandrutu I mentioned that possibility in the issue. 

I think we do need a String method. I also think making 'IsMatch' a little smarter so we don't always have to call 'String' when we use 'IsMatch' is worth it.

Reply:
Ok... make sure we have the same logic as in https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/pcommon/common.go#L434

Reply:
@bogdandrutu I've updated the function to handle simple cases quickly, and defer to pdata in more complicated cases.

Reply:
My point was for the "simple" cases to be consistent with what AsString does. I don't think we want to do what you did on the default for the moment.

Reply:
Ok I've updated the switch statement to handle what I think is a reasonable amount of situations based on what the accessors return and the grammar allows.  I left the `float64` the same as the existing `attributematcher` logic.